ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/CommonHttpClientTest.kt

```diff
@@ -36,4 +36,26 @@ class CommonHttpClientTest {
         // When the engine is provided by Ktor factory is should be closed together with the client.
         assertFalse { engine.isActive }
     }
+
+    @Test
+    fun testHttpClientClosesInstalledFeatures() {
+        val client = HttpClient(MockEngine) {
+            engine { addHandler { respond("") } }
+            install(TestFeature)
+        }
+        client.close()
+        assertTrue(client.feature(TestFeature)!!.closed)
+    }
+    class TestFeature : Closeable {
+        var closed = false
```

Comment:
Native tests are failing on mutating this property. You can use `var closed by shared(false)` to fix it.